plugins/carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -5,24 +5,19 @@
 import requests
 from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
 import time
-import icon_carbon_black_cloud.util.agent_typer as agent_typer
-import urllib.parse
+from icon_carbon_black_cloud.util import agent_typer
+from icon_carbon_black_cloud.util.constants import DEFAULT_TIMEOUT
 
 
 class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
 
     def connect(self, params):
-        self.api_id = params.get(Input.API_ID)
-        self.api_secret = params.get(Input.API_SECRET_KEY).get("secretKey")
-
-        base_url = params.get(Input.URL)
-        self.base_url = urllib.parse.urlparse(base_url).hostname
-        self.base_url = f"https://{self.base_url}"
-
-        self.org_key = params.get(Input.ORG_KEY)
-
+        self.api_id = params.get(Input.API_ID, "")
+        self.api_secret = params.get(Input.API_SECRET_KEY, {}).get("secretKey", "")
+        self.org_key = params.get(Input.ORG_KEY, "")
+        self.base_url = params.get(Input.URL, "")
```

Comment:
Should we still need the url to be provided or is it always a fixed url?

Reply:
It seems there are more regions. It depends on which `URL` customer logs in. I added all the possibilities as an enum 😄